Sources/Shared/AuthenticationController.swift

```diff
@@ -166,29 +177,38 @@ final public class AuthenticationController {
         return URI
     }
     
-    /**
-     Generate a URL to open the Vimeo code grant authorization page.  When opened in Safari, this page allows users to log into your application.
-     
-     - returns: the code grant authorization page URL
-     */
-    public func codeGrantAuthorizationURL() -> URL {
+    /// Generate a URL to open the Vimeo code grant authorization page.  When opened in Safari, this page allows users to log into your application.
+    ///
+    /// - Parameters:
+    ///   - name: Optional placeholder: when set, the user will see a placeholder name inside the name text field in Safari
+    ///   - email: Optional placeholder: when set, the user will see a placeholder email inside the email text field in Safari
+    ///   - action: Set this parameter to send user straight to join or login inside Safari, default is login
+    /// - Returns: the code grant authorization page URL
+    public func codeGrantAuthorizationURL(usingPlaceholdersForName name: String? = nil, andEmail email: String? = nil, showingInitialAuthorizationAction action: AuthorizationAction = .login) -> URL {
         let parameters = [Constants.ResponseTypeKey: Constants.CodeKey,
                           Constants.ClientIDKey: self.configuration.clientIdentifier,
                           Constants.RedirectURIKey: self.codeGrantRedirectURI,
                           Constants.ScopeKey: Scope.combine(self.configuration.scopes),
                           Constants.StateKey: type(of: self).state]
         
-        let urlString = self.configuration.baseUrl.appendingPathComponent(Constants.CodeGrantAuthorizationPath).absoluteString
+        let urlString = self.configuration.baseUrl.appendingPathComponent(Constants.CodeGrantAuthorizationPath)
+            .appendingQueries([
+                URLQueryItem(name: Constants.URLQueries.Email, value: email),
+                URLQueryItem(name: Constants.URLQueries.Name, value: name),
+                URLQueryItem(name: Constants.URLQueries.AuthorizationAction, value: action.rawValue)
```

Comment:
Because we always have a default value for `login` this is changing the current behavior in that the URL returned from `codeGrantAuthorization()` will now always include the `auth_action` query item. 

How can we ensure that this change will not affect the existing authorization flow?

Reply:
Agree, we need to let the user choose what to add.